### PR TITLE
Feat: Implement no-duplicate-imports error rule

### PR DESCRIPTION
**Pull request by thecp**
refs #53

### CODE REVIEW

File: crates/rslint_core/src/groups/errors/no_duplicate_imports.rs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    ```
+
+    ## Including Exports
+
+    ### Incorrect Code Examples
+
+    ```js
+    import { foo } from "bla";
+    export { foo } from "bla";
+    ```
```
Comment (Stupremee) [inline crates/rslint_core/src/groups/errors/no_duplicate_imports.rs:26]: Can you squash all of these codeblocks into a single one and write a comment "including exports"

File: crates/rslint_parser/src/ast/stmt_ext.rs [hunk 2]
```diff
@@ -253,13 +253,24 @@ impl VarDecl {
     }
 }
 
-impl ImportDecl {
+pub trait Source {
```
Comment (Stupremee) [inline crates/rslint_parser/src/ast/stmt_ext.rs:256]: I think it's better to add those methods on the structs instead of introducing a new trait.

**Review by Stupremee — CHANGES_REQUESTED** [review]

**Comment by Stupremee** [comment]
Two small nits. Otherwise looks really good! Thank you

**Comment by thecp** [comment]
Thx @Stupremee all suggestions should be addressed

**Review by Stupremee — APPROVED** [review]

**Comment by Stupremee** [comment]
Hmm. Seems like a rustdoc regression. I will investigate a bit. Temporarily, you can put `doc = false` into the `rslint-lsp` binary.

**Pull request merged**